### PR TITLE
allow to config font

### PR DESCRIPTION
**Pull request by Darmody**
Thanks for `letter_avatar` which make my life easier!

The default fonts can't support Chinese font, So I think font may be a configurable part.

### CODE REVIEW

File: lib/letter_avatar.rb [hunk 1]
```diff
@@ -5,7 +5,9 @@
 module LetterAvatar
 
   mattr_accessor :cache_base_path
+  mattr_accessor :font
   @@cache_base_path = nil
+  @@font = nil
```
Comment (Darmody) [inline lib/letter_avatar.rb]: updated. May I ask why?

**Comment by huacnlee** [comment]
Chinese chars are complex, render them on the image may not beautiful.

You can covert Chinese use Pinyin first letter like this:

``` rb
class User
  def username_for_avatar
    Pinyin.convert(self.name)[0]
  end
end
```

And then:

``` rb
letter_avatar_for(user.username_for_avatar, 120)
```

**Comment by ksz2k** [comment]
but why shouldn't we allow to configure the font?

File: lib/letter_avatar.rb [hunk 2]
```diff
@@ -5,7 +5,9 @@
 module LetterAvatar
 
   mattr_accessor :cache_base_path
+  mattr_accessor :font
   @@cache_base_path = nil
+  @font = nil
```
Comment (ksz2k) [inline lib/letter_avatar.rb]: please, use `@@`, as it is in other config options

File: lib/letter_avatar/avatar.rb [hunk 3]
```diff
@@ -31,6 +31,10 @@ def cache_path
         "#{LetterAvatar.cache_base_path || 'public/system'}/letter_avatars/#{VERSION}"
       end
 
+      def font
+        "#{LetterAvatar.font || 'Roboto-Medium'}"
+      end
```
Comment (ksz2k) [inline lib/letter_avatar/avatar.rb:36]: also, please remove this method.. IMHO it is ok just to set this Roboto-Medium directly in `LetterAvatar.font`

File: lib/letter_avatar/avatar.rb [hunk 4]
```diff
@@ -70,7 +74,7 @@ def generate_fullsize(identity)
           -size 240x240
           xc:#{to_rgb(color)}
           -pointsize 140
-          -font Roboto-Medium
+          -font #{font}
```
Comment (ksz2k) [inline lib/letter_avatar/avatar.rb:77]: and use `LetterAvatar.font` here, after setting that up to Roboto default

**Comment by huacnlee** [comment]
Can custom font will get new issues: draw position, font size...

**Comment by ksz2k** [comment]
And, last but not least..

> Thanks for letter_avatar which make my life easier!

nice to hear that :-)

**Comment by ksz2k** [comment]
You're right @huacnlee 

Hmm, ok so won't merge - sorry..

**Comment by huacnlee** [comment]
By default font `Roboto-Medium` there have a position fix: `-annotate -0+10`, if user change to other font, the draw position may not in absolute center.

**Comment by Darmody** [comment]
I agree with @huacnlee .

Then I suggest that changing another default font which can support more characters.

**Comment by huacnlee** [comment]
1. There no font support all chars, except Arial, Helvetica, but that not good.
2. Why don't you use other fields to generate the avatar? For example: email
3. I have give you a solution to support Chinese chars: Translate them into Pinyin.

**Comment by Darmody** [comment]
@huacnlee  Thanks for your advices. 

I need to generate the avatar by Chinese letter indeed.

Nevermind, I'll find another solution.

**Pull request closed without merging**